File: audit/src/main/java/tech/beshu/ror/commons/ResponseContext.java
package tech.beshu.ror.commons;


import tech.beshu.ror.commons.shims.request.RequestContextShim;

/**
 * Created by sscarduzio on 28/06/2017.
 */
@Deprecated
public class ResponseContext {

  private final FinalState finalState;
  private final Long finishedHandling;
  private final RequestContextShim rc;
  private final Throwable error;
  private final Verbosity verbosity;
  private final String reason;
  private final boolean isMatch;

  public ResponseContext(FinalState finalState, RequestContextShim rc, Throwable error, Verbosity verbosity, String reason, boolean isMatch) {
    this.finalState = finalState;
    this.finishedHandling = System.currentTimeMillis();
    this.rc = rc;
    this.verbosity = verbosity;
    this.error = error;
    this.reason = reason;
    this.isMatch = isMatch;
  }

  public boolean getIsMatch() {
    return isMatch;
  }

  public String getReason() {
    return reason;
  }

  public Long getDurationMillis() {
    return finishedHandling - rc.getTimestamp().getTime();
  }

  public Throwable getError() {
    return error;
  }

  public Verbosity getVerbosity() {
    return verbosity;
  }

  public RequestContextShim getRequestContext() {
    return rc;
  }

  public FinalState finalState() {
    return finalState;
  }

  public enum FinalState {
    FORBIDDEN,
    ALLOWED,
    ERRORED,
    NOT_FOUND
  }
}

File: corex/src/main/scala/tech/beshu/ror/acl/logging/ResponseContext.scala
package tech.beshu.ror.acl.logging

import tech.beshu.ror.acl.blocks.{Block, BlockContext}
import tech.beshu.ror.acl.request.RequestContext


sealed trait ResponseContext {
  def requestContext: RequestContext
}
object ResponseContext {

  final case class Allowed(requestContext: RequestContext,
                           block: Block,
                           blockContext: BlockContext,
                           history: Vector[Block.History])
    extends ResponseContext

  final case class ForbiddenBy(requestContext: RequestContext,
                               block: Block,
                               blockContext: BlockContext,
                               history: Vector[Block.History])
    extends ResponseContext

  final case class Forbidden(requestContext: RequestContext,
                             history: Vector[Block.History])
    extends ResponseContext

  final case class Errored(requestContext: RequestContext,
                           cause: Throwable)
    extends ResponseContext

  final case class NotFound(requestContext: RequestContext,
                            cause: Throwable)
    extends ResponseContext

}
